# DP: aarch64-simd: Avoid emitting dup insn by using canonical form for fnma.

--- a/src/gcc/config/aarch64/aarch64-simd.md
+++ b/src/gcc/config/aarch64/aarch64-simd.md
@@ -1674,9 +1674,8 @@
 (define_insn "fnma<mode>4"
   [(set (match_operand:VHSDF 0 "register_operand" "=w")
 	(fma:VHSDF
-	  (match_operand:VHSDF 1 "register_operand" "w")
-          (neg:VHSDF
-	    (match_operand:VHSDF 2 "register_operand" "w"))
+	  (neg:VHSDF (match_operand:VHSDF 1 "register_operand" "w"))
+	  (match_operand:VHSDF 2 "register_operand" "w")
 	  (match_operand:VHSDF 3 "register_operand" "0")))]
   "TARGET_SIMD"
   "fmls\\t%0.<Vtype>, %1.<Vtype>, %2.<Vtype>"
